core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -634,6 +636,10 @@ boolean isPreserveAliases() {
                 if (!noIndicesAllowedOrMatches(options, matches)) {
                     throw infe(expression);
                 }
+
+                if (expression.contains("*")) {
+                    wildcardSeen = true;
```

Comment:
I wonder if we should set this to true only if options.expandWildcardsOpen || options.expandWildcardsClosed . Otherwise wildcard expressions don't get expanded...